### PR TITLE
DrivAerML: champion recovery seed=7 no-compile

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Re-run the exact champion config (ncl1dh88) with seed=7 and --no-compile-model. Second seed variant alongside jet (seed=42) and norman (seed=0) to map seed sensitivity distribution.

## Bucket
A — Champion Recovery / Reproducibility

## Command
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --no-compile-model --seed 7 --wandb_group dm-champion-recovery
```

## Kill Condition
If best truncated val still above 8% after epoch 120, kill.

## Baseline
- **val surface_rel_l2_pct:** 3.833% at epoch 511
- **test surface_rel_l2_pct:** 4.685%
- **W&B run:** ncl1dh88 (eren/dm-ema-9995-gc05)
- **Full-eval test evidence:** ~4.39-4.50% (see issue #3283)
- **Target:** <3.71% (AB-UPT benchmark)

## Expected outcome
With 3 seeds (0, 7, 42) running in parallel we will establish the full seed distribution and identify the best seed for further scaling.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT megumi:

## Results

### Champion Recovery — seed=7, no-compile (run `5utwhkv0`)

| Metric | Baseline (ncl1dh88) | Seed=7 | Delta |
|--------|---------------------|--------|-------|
| Best val surface_rel_l2_pct | 3.833% @ep511 | **3.832% @ep867** | -0.03% (matches) |
| Test surface_rel_l2_pct | 4.685% | **4.832%** | +3.1% worse |
| Epochs trained | 511 (of longer run) | 874 (600-min timeout) | — |
| Epoch time | ~22s (baseline hardware) | ~41s (RTX PRO 6000 Blackwell, no-compile) | — |

**Note:** Test metric is from truncated eval (`--max-eval-batches 200`). Full-eval test evidence from issue #3283 suggests ~4.39-4.50% for the baseline; this run's truncated test of 4.832% is consistent with the truncation gap. A full-eval from the best checkpoint would give the true test number.

**Convergence trajectory:**
```
ep   1: val=61.42%  best=61.42%
ep  30: val=17.58%  best=17.58%
ep 100: val= 7.12%  best= 7.12%
ep 120: val= 6.47%  best= 6.47%   ← kill gate passed (< 8%)
ep 200: val= 5.01%  best= 4.99%
ep 300: val= 4.52%  best= 4.51%
ep 400: val= 4.34%  best= 4.34%
ep 500: val= 4.17%  best= 4.15%
ep 600: val= 4.03%  best= 4.03%
ep 700: val= 4.02%  best= 3.96%
ep 800: val= 3.91%  best= 3.89%
ep 867: val= 3.83%  best= 3.83%   ← BEST (matches baseline)
ep 874: val= 3.88%  best= 3.83%   ← final (timeout)
```

**Command:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 \
  --model-heads 8 --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 \
  --no-compile-model --seed 7 \
  --wandb-group dm-champion-recovery \
  --wandb-name "megumi/dm-champion-seed7" --agent megumi
```

## What happened

**Seed=7 successfully reproduces the champion baseline.** Best val of 3.832% matches the baseline's 3.833% within noise. This confirms the champion config is robust across seeds (at least for seed=7).

Key observations:
- **Convergence is smooth and monotonic** — no catastrophic LR restart instability (unlike MoE experiments). The cosine annealing with T_max=30 works well for the standard architecture.
- **Best val reached at ep867** vs baseline's ep511 — seed=7 converges more slowly but reaches the same final quality. The longer epoch count was made possible by the 600-min timeout and ~41s/epoch (no-compile on RTX PRO 6000 Blackwell).
- **Test metric (4.832%, truncated eval)** needs full-eval to be comparable. The best checkpoint is saved and ready for full evaluation.

## Suggested follow-ups

- **Full-eval the best checkpoint** (ep867) without `--max-eval-batches` to get the true test metric for paper comparison.
- **Compare across seeds** (0, 7, 42) once jet and norman complete their runs — the seed distribution will show whether 3.83% is the floor or if some seeds go lower.
- **This checkpoint is a candidate for fine-tuning** with metric-aware loss (per issue #3283 breakout lane 2) since it's at the champion quality level.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3286 because ADVISOR: Closing — seed=7 reproduces champion val (3.832% vs 3.833%) but does not improve. Batch-limited test 4.832% is slightly worse than baseline 4.685%. This confirms the champion config ceiling is ~3.83% val across seeds — meaning further seed runs will converge to the same point. Breaking past this ceiling requires a config change, not more seeds. Megumi reassigned to a novel DM direction (weight decay exploration)..
